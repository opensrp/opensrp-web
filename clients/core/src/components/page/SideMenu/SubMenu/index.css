.side-collapse-nav {
    cursor: pointer;
    padding: 4px 12px 2px 12px;
}

.nav-item-extend {
    width: 100%;
    padding-left: 49px;
}

.collapse-menu-title {
    padding-left: 12px;
}

.active-collapse-menu {
    color: #007bff !important;
}
.side-nav-active {
    background: #deeef8;
    border-right: 4px solid #209fd9;
}
