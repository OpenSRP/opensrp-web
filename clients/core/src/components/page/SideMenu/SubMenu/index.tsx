import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React, { Fragment } from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import { NavLink } from 'react-router-dom';
import { Collapse, Nav, NavItem } from 'reactstrap';
import './index.css';

/** interface for a module i.e group of related pages */
export interface NavigationModule {
  /** Page navigation for the module, collapses to reveal the page links */
  parentNav: ModulePageLink;
  /** pageLinks to pages under this navigational module */
  childNavs: PageLink[];
}

/** interface for objs describing links to different pages */
export interface PageLink {
  /** the link's displayable text */
  label: string;
  /** URL to get redirected on click */
  url: string;
}

/** interface for the module page links, module represents collection of related page links */
export interface ModulePageLink {
  /** text label visible on menu item */
  label: string;
  /** fontAwesome icon displayed on menu item */
  icon: IconProp;
  /** url that module link will redirect to */
  url?: string;
}

/** props for this component */
export interface SubMenuProps {
  /** object describing navigation data for the module page link */
  parentNav: ModulePageLink;
  /** array of the pageNavigation links grouped under this navigation module */
  childNavs: PageLink[];
  /** string literal with the label value of currently collapsed module navigation link */
  collapsedModuleLabel: string | null;
  /** callback to parent component changes the collapsedModuleLabel
   *  entry in the parent component
   */
  setCollapsedModuleLabel?: SetCollapsedModuleLabel;
}

/** this components state */
export interface SubMenuState {
  /** whether the child navigation links should be collapsed or hidden */
  isActiveModule: boolean;
}

/** takes a string and updates the current collapsed-module-label in the parent component
 * @param {string} label - label of clicked navigation module page link
 */
type SetCollapsedModuleLabel = (label: string) => void;

/** Default props */
export const defaultSubMenuProps: SubMenuProps = {
  childNavs: [],
  collapsedModuleLabel: '',
  parentNav: {
    icon: ['far', 'user'],
    label: 'home',
  },
};

/** intersection of all types describing the props */
type subMenuPropsTypes = SubMenuProps & RouteComponentProps;

/** The SubMenu component */
export class SubMenu extends React.Component<subMenuPropsTypes, SubMenuState> {
  public static defaultProps = defaultSubMenuProps;

  public render() {
    const { childNavs, parentNav, collapsedModuleLabel, location } = this.props;
    const pathName = location.pathname;
    /** whether to collapse the child pages for this navigation module */
    const collapseMenu: boolean = collapsedModuleLabel === parentNav.label;
    /** whether the navigation module is currently active; judging from the history.location pathname
     * set to true if one of the module navigation's children has a url similar to pathname or
     * if the url of the parentNav is similar to the pathname
     */
    const isActiveModule: boolean =
      childNavs.map(nav => nav.url).indexOf(pathName!) > -1 || parentNav.url === pathName;

    const moduleLinkJsx = (
      <Fragment>
        <FontAwesomeIcon icon={parentNav.icon} size="lg" />
        <span className="collapse-menu-title"> {parentNav.label} </span>
      </Fragment>
    );

    const moduleLinkClassName = classNames('nav', 'nav-link', 'side-nav-item', {
      'active-collapse-menu': isActiveModule,
    });

    return (
      <div id={`sub-menu-${parentNav.label.replace(' ', '-')}`}>
        <Nav
          id={parentNav.label.replace(' ', '-')}
          className="side-collapse-nav"
          onClick={this.setModuleLabel}
        >
          <NavItem id="module-link">
            {parentNav.url ? (
              <NavLink to={parentNav.url} className={moduleLinkClassName}>
                {moduleLinkJsx}
              </NavLink>
            ) : (
              <span className={moduleLinkClassName}>{moduleLinkJsx}</span>
            )}
          </NavItem>
        </Nav>

        <Collapse isOpen={collapseMenu}>
          {childNavs.map((childNavObj, key) => {
            return (
              <Nav key={'child-nav-' + key}>
                <NavItem className="nav-item-extend">
                  <NavLink
                    to={childNavObj.url}
                    className="nav-link side-nav-item"
                    activeClassName="side-nav-active"
                  >
                    <span> {childNavObj.label} </span>
                  </NavLink>
                </NavItem>
              </Nav>
            );
          })}
        </Collapse>
      </div>
    );
  }

  /** updates parent component with the label of the currently collapsed navigation module */
  private setModuleLabel = () => {
    const { setCollapsedModuleLabel, collapsedModuleLabel, parentNav } = this.props;

    const isLabelSet = collapsedModuleLabel === parentNav.label;
    // if label is already set then this will behave as a toggle
    isLabelSet && setCollapsedModuleLabel !== undefined
      ? setCollapsedModuleLabel('')
      : setCollapsedModuleLabel!(this.props.parentNav.label);
  };
}

export default withRouter(SubMenu);
