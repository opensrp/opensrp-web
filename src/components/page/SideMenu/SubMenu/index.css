.side-collapse-nav.nav {
  cursor: pointer;
  padding: 4px 12px 2px 20px;
  max-width: initial !important;
}

.nav-item-extend {
  width: 100%;
  padding-left: 20px;
}

.collapse-menu-title {
  padding-left: 12px;
}

.active-collapse-menu {
  color: #0192d4 !important;
}

.active-collapse-menu:hover {
  color: #0192d4 !important;
}

.side-nav-active {
  border-right: 4px solid #209fd9;
}

.side-menu-container .side-menu-extend ul.nav .collapse.show {
  padding-right: initial;
  padding-left: 33px;
}

.side-nav-item.nav {
  max-width: initial;
  background-color: initial;
  font-family: 'Roboto';
  font-size: 14px;
  font-weight: 500;
}

.side-nav-item.nav:hover {
  color: #0192d4 !important;
}

.side-menu-container .collapse {
  padding-right: initial;
  padding-left: 33px;
}

.active-collapse-menu path {
  fill: #0192d4;
}

path {
  fill: #1b314b;
}
