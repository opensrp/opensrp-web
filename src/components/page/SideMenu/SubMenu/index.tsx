import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import classNames from 'classnames';
import React, { Fragment } from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import { NavLink } from 'react-router-dom';
import { Collapse, Nav, NavItem } from 'reactstrap';
import './index.css';

/** interface for a module i.e group of related pages
 * @property {string} parentNav - Page navigation for the module, collapses to reveal the page links
 * @property {PageLink[]} childNavs - pageLinks to pages under this navigational module
 */
export interface NavigationModule {
  parentNav: ModulePageLink;
  childNavs: PageLink[];
}

/** interface for objs describing links to different pages
 * @property {string} label - the link's displayable text
 * @property {string} url - URL to get redirected on click
 */
export interface PageLink {
  label: string;
  url: string;
}

/** interface for the module page links, module represents collection of related page links
 * @property {string} label - text label visible on menu item
 * @property {IconProp} icon- fontAwesome icon displayed on menu item
 * @property {string} url - url that module link will redirect to
 */
export interface ModulePageLink {
  label: string;
  icon: IconProp;
  url?: string;
}

/** props for this component
 * @property {ModulePageLink} parentNav - object describing navigation data for the module page link
 * @property {PageLink[]} childNavs - array of the pageNavigation links grouped under this navigation module
 * @property {string} collapsedModuleLabel - string literal with the label value of currently collapsed module navigation link
 * @property {setCollapsedModuleLabel} setCollapsedModuleLabel - callback to parent component;
 *            changes the collapsedModuleLabel entry in the parent component
 */
export interface SubMenuProps {
  parentNav: ModulePageLink;
  childNavs: PageLink[];
  collapsedModuleLabel: string;
  setCollapsedModuleLabel?: SetCollapsedModuleLabel;
  customIcon?: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
}

/** this components state
 * @property {boolean} collapseMenu - whether the child navigation links should be collapsed or hidden
 */
export interface SubMenuState {
  isActiveModule: boolean;
}

/** takes a string and updates the current collapsed-module-label in the parent component
 * @param {string} label - label of clicked navigation module pagelink
 */
type SetCollapsedModuleLabel = (label: string) => void;

/** Default props */
export const defaultSubMenuProps: SubMenuProps = {
  childNavs: [],
  collapsedModuleLabel: '',
  parentNav: {
    icon: ['far', 'user'],
    label: 'home',
  },
};

/** intersection of all types describing the props */
type subMenuPropsTypes = SubMenuProps & RouteComponentProps;

export class SubMenu extends React.Component<subMenuPropsTypes, SubMenuState> {
  public static defaultProps = defaultSubMenuProps;

  public render() {
    const { childNavs, parentNav, collapsedModuleLabel, location } = this.props;
    const pathName = location.pathname;
    /** whether to collapse the child pages for this navigation module */
    const collapseMenu: boolean = collapsedModuleLabel === parentNav.label;
    /** whether the navigation module is currently active; judging from the history.location pathname
     * set to true if one of the module navigation's children has a url similar to pathname or
     * if the url of the parentNav is similar to the pathname
     */
    const isActiveModule: boolean =
      childNavs.map(nav => nav.url).indexOf(pathName!) > -1 || parentNav.url === pathName;

    const moduleLinkJsx = (
      <Fragment>
        {this.props.customIcon ? (
          <this.props.customIcon />
        ) : (
          <FontAwesomeIcon icon={parentNav.icon} size="lg" />
        )}
        <span className="collapse-menu-title"> {parentNav.label} </span>
      </Fragment>
    );

    const moduleLinkClassName = classNames('nav', 'nav-link', 'side-nav-item', {
      'active-collapse-menu': isActiveModule,
    });

    return (
      <div id={`sub-menu-${parentNav.label.replace(' ', '-')}`}>
        <Nav
          id={parentNav.label.replace(' ', '-')}
          className="side-collapse-nav"
          onClick={this.setModuleLabel}
        >
          <NavItem>
            {parentNav.url ? (
              <NavLink to={parentNav.url} className={moduleLinkClassName}>
                {moduleLinkJsx}
              </NavLink>
            ) : (
              <span className={moduleLinkClassName}>{moduleLinkJsx}</span>
            )}
          </NavItem>
        </Nav>

        <Collapse isOpen={collapseMenu}>
          {childNavs.map((childNavObj, key) => {
            return (
              <Nav key={'child-nav-' + key}>
                <NavItem className="nav-item-extend">
                  <NavLink
                    to={childNavObj.url}
                    className="nav-link side-nav-item"
                    activeClassName="side-nav-active"
                  >
                    <span> {childNavObj.label} </span>
                  </NavLink>
                </NavItem>
              </Nav>
            );
          })}
        </Collapse>
      </div>
    );
  }

  /** updates parent component with the label of the currently collapsed navigation module */
  private setModuleLabel = () => {
    const labelAlreadySet = this.props.collapsedModuleLabel === this.props.parentNav.label;
    labelAlreadySet && this.props.setCollapsedModuleLabel !== undefined
      ? this.props.setCollapsedModuleLabel('')
      : this.props.setCollapsedModuleLabel!(this.props.parentNav.label);
  };
}

export default withRouter(SubMenu);
