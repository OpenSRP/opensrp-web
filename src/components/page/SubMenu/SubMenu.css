.side-collapse-nav {
  cursor: pointer;
}

.nav-item-extend {
  width: 100%;
}
