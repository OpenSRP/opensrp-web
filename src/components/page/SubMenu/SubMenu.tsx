import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React from 'react';
import { NavLink } from 'react-router-dom';
import { Collapse, Nav, NavItem } from 'reactstrap';
import { CLIENT_NAVIGATION_MODULE_OBJECT } from '../../../constants';
import './SubMenu.css';

export interface NavObj {
  navLabel: string;
  navURL: string;
}

export interface NavCollapseObj {
  navLabel: string;
  navIcon: IconProp;
}

export interface SubMenuProps {
  identifier: string;
  isCollapseMenuActive: boolean;
  isExpand: boolean;
  parentNav: NavCollapseObj;
  childNavs: NavObj[];
  setSideMenuActive: any;
  setSideMenuToggle: any;
}

/** By default load Clients Sub Menu */
export const defaultSubMenuProps: SubMenuProps = {
  ...CLIENT_NAVIGATION_MODULE_OBJECT,
  isCollapseMenuActive: false,
  isExpand: true,
  setSideMenuActive: null,
  setSideMenuToggle: null,
};

class SubMenu extends React.Component<SubMenuProps, {}> {
  public static defaultProps = defaultSubMenuProps;
  constructor(props: SubMenuProps) {
    super(props);
  }

  public render() {
    const { childNavs, parentNav } = this.props;
    return (
      <div>
        <Nav>
          <NavItem onClick={this.toggle}>
            <a
              className={classNames('nav', 'nav-link', 'side-nav-item', {
                'active-collapse-menu': this.props.isCollapseMenuActive,
              })}
            >
              <FontAwesomeIcon icon={parentNav.navIcon} />
              <span> {parentNav.navLabel} </span>
            </a>
          </NavItem>
        </Nav>
        <Collapse isOpen={this.props.isExpand}>
          {childNavs.map((childNavObj, key) => {
            return (
              <Nav key={'child-nav-' + key}>
                <NavItem>
                  <NavLink
                    to={childNavObj.navURL}
                    className="nav-link side-nav-item"
                    activeClassName="side-nav-active"
                    onClick={this.setActiveClassNameToCollapse}
                  >
                    <span> {childNavObj.navLabel} </span>
                  </NavLink>
                </NavItem>
              </Nav>
            );
          })}
        </Collapse>
      </div>
    );
  }

  private toggle = () => {
    this.props.setSideMenuToggle(this.props.identifier);
  };

  private setActiveClassNameToCollapse = () => {
    this.props.setSideMenuActive(this.props.identifier);
  };
}

export default SubMenu;
