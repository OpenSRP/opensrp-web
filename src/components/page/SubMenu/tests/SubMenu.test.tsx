import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import SubMenu, { SubMenuProps } from '../SubMenu';

const history = createBrowserHistory();

describe('components/page/SideMenu', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props: SubMenuProps = {
      childNavs: [],
      identifier: 'CLIENT',
      isCollapseMenuActive: true,
      isExpand: true,
      parentNav: { navIcon: ['far', 'user'], navLabel: 'All client Records' },
      setSideMenuActive: null,
      setSideMenuToggle: null,
    };
    shallow(
      <Router history={history}>
        <SubMenu {...props} />
      </Router>
    );
  });

  it('renders expanded nav correctly', () => {
    const props: SubMenuProps = {
      childNavs: [{ navLabel: 'Users', navURL: '/404' }, { navLabel: 'Roles', navURL: '/404' }],
      identifier: 'ADMIN',
      isCollapseMenuActive: true,
      isExpand: true,
      parentNav: { navIcon: ['fas', 'cog'], navLabel: 'Admin' },
      setSideMenuActive: null,
      setSideMenuToggle: null,
    };
    const wrapper = mount(
      <Router history={history}>
        <SubMenu {...props} />
      </Router>
    );
    expect(toJson(wrapper.find('SubMenu'))).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders collapsed nav correctly', () => {
    const props: SubMenuProps = {
      childNavs: [{ navLabel: 'Users', navURL: '/404' }],
      identifier: 'ADMIN',
      isCollapseMenuActive: true,
      isExpand: false,
      parentNav: { navIcon: ['fas', 'cog'], navLabel: 'Admin' },
      setSideMenuActive: null,
      setSideMenuToggle: null,
    };
    const wrapper = mount(
      <Router history={history}>
        <SubMenu {...props} />
      </Router>
    );
    expect(toJson(wrapper.find('SubMenu'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
