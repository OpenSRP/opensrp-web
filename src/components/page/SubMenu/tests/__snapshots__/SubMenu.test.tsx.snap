// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/SubMenu renders collapsed nav correctly 1`] = `
<SubMenu
  childNavs={
    Array [
      Object {
        "navLabel": "Users",
        "navURL": "/404",
      },
    ]
  }
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  identifier="ADMIN"
  isCollapseMenuActive={true}
  isExpand={false}
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  parentNav={
    Object {
      "navIcon": Array [
        "fas",
        "cog",
      ],
      "navLabel": "Admin",
    }
  }
>
  <div>
    <Nav
      className="side-collapse-nav"
      onClick={[Function]}
      tag="ul"
      vertical={false}
    >
      <ul
        className="side-collapse-nav nav"
        onClick={[Function]}
      >
        <NavItem
          tag="li"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav nav-link side-nav-item"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "cog",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="lg"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-cog fa-w-16 fa-lg "
                  data-icon="cog"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <span
                className="collapse-menu-title"
              >
                 
                Admin
                 
              </span>
            </a>
          </li>
        </NavItem>
      </ul>
    </Nav>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={350}
        unmountOnExit={false}
      >
        <div
          className="collapse"
          style={Object {}}
        >
          <Nav
            key="child-nav-0"
            tag="ul"
            vertical={false}
          >
            <ul
              className="nav"
            >
              <NavItem
                className="nav-item-extend"
                tag="li"
              >
                <li
                  className="nav-item-extend nav-item"
                >
                  <NavLink
                    activeClassName="side-nav-active"
                    className="nav-link side-nav-item"
                    to="/404"
                  >
                    <Link
                      aria-current={null}
                      className="nav-link side-nav-item"
                      to="/404"
                    >
                      <a
                        aria-current={null}
                        className="nav-link side-nav-item"
                        href="/404"
                        onClick={[Function]}
                      >
                        <span>
                           
                          Users
                           
                        </span>
                      </a>
                    </Link>
                  </NavLink>
                </li>
              </NavItem>
            </ul>
          </Nav>
        </div>
      </Transition>
    </Collapse>
  </div>
</SubMenu>
`;

exports[`components/page/SubMenu renders expanded nav correctly 1`] = `
<SubMenu
  childNavs={
    Array [
      Object {
        "navLabel": "Users",
        "navURL": "/404",
      },
      Object {
        "navLabel": "Roles",
        "navURL": "/404",
      },
    ]
  }
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  identifier="ADMIN"
  isCollapseMenuActive={true}
  isExpand={true}
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  parentNav={
    Object {
      "navIcon": Array [
        "fas",
        "cog",
      ],
      "navLabel": "Admin",
    }
  }
>
  <div>
    <Nav
      className="side-collapse-nav"
      onClick={[Function]}
      tag="ul"
      vertical={false}
    >
      <ul
        className="side-collapse-nav nav"
        onClick={[Function]}
      >
        <NavItem
          tag="li"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav nav-link side-nav-item"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "cog",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="lg"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-cog fa-w-16 fa-lg "
                  data-icon="cog"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <span
                className="collapse-menu-title"
              >
                 
                Admin
                 
              </span>
            </a>
          </li>
        </NavItem>
      </ul>
    </Nav>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={350}
        unmountOnExit={false}
      >
        <div
          className="collapse show"
          style={Object {}}
        >
          <Nav
            key="child-nav-0"
            tag="ul"
            vertical={false}
          >
            <ul
              className="nav"
            >
              <NavItem
                className="nav-item-extend"
                tag="li"
              >
                <li
                  className="nav-item-extend nav-item"
                >
                  <NavLink
                    activeClassName="side-nav-active"
                    className="nav-link side-nav-item"
                    to="/404"
                  >
                    <Link
                      aria-current={null}
                      className="nav-link side-nav-item"
                      to="/404"
                    >
                      <a
                        aria-current={null}
                        className="nav-link side-nav-item"
                        href="/404"
                        onClick={[Function]}
                      >
                        <span>
                           
                          Users
                           
                        </span>
                      </a>
                    </Link>
                  </NavLink>
                </li>
              </NavItem>
            </ul>
          </Nav>
          <Nav
            key="child-nav-1"
            tag="ul"
            vertical={false}
          >
            <ul
              className="nav"
            >
              <NavItem
                className="nav-item-extend"
                tag="li"
              >
                <li
                  className="nav-item-extend nav-item"
                >
                  <NavLink
                    activeClassName="side-nav-active"
                    className="nav-link side-nav-item"
                    to="/404"
                  >
                    <Link
                      aria-current={null}
                      className="nav-link side-nav-item"
                      to="/404"
                    >
                      <a
                        aria-current={null}
                        className="nav-link side-nav-item"
                        href="/404"
                        onClick={[Function]}
                      >
                        <span>
                           
                          Roles
                           
                        </span>
                      </a>
                    </Link>
                  </NavLink>
                </li>
              </NavItem>
            </ul>
          </Nav>
        </div>
      </Transition>
    </Collapse>
  </div>
</SubMenu>
`;
